### PR TITLE
Fix grammar bitmask shape mismatch with padded vocab logits

### PR DESCRIPTION
**Pull request by sraizada-tt**
## Summary
- When models pad lm_head to `padded_vocab_size` for on-device sampling (e.g. GPT-OSS-120B pads from 201088 to 262144 for pow2 topk), the logits tensor is wider than `vocab_size`
- The grammar bitmask unpacked from structured output is only `vocab_size` wide, causing `masked_fill_` to crash with shape mismatch: `The size of tensor a (262144) must match the size of tensor b (201088)`
- Fix: pad the bitmask with `True` (mask out) for positions beyond `vocab_size` so padding tokens are always set to `-inf`

vllm nightly for gpt-oss: https://github.com/tenstorrent/tt-metal/actions/runs/22435248307

### CODE REVIEW

File: vllm/v1/worker/tt_model_runner.py [hunk 1]
```diff
@@ -1727,9 +1727,17 @@ def apply_grammar_bitmask(
...
+        vocab_logits = logits.shape[-1]
+        vocab_bitmask = unpacked_bitmask.shape[-1]
+        if vocab_bitmask < vocab_logits:
+            # Logits may be wider than vocab (e.g. padded for on-device sampling).
+            # Pad bitmask with True (mask out) so padding positions are set to -inf.
+            unpacked_bitmask = torch.nn.functional.pad(
+                unpacked_bitmask, (0, vocab_logits - vocab_bitmask), value=True
+            )
+        elif vocab_bitmask > vocab_logits:
+            unpacked_bitmask = unpacked_bitmask[:, :vocab_logits]
```
Comment (skhorasganiTT) [inline vllm/v1/worker/tt_model_runner.py:1740]: I don't think we should be adding torch padding / handling this on the vLLM side. Can the model instead slice the logits and return the correct output shape? That should also be more efficient. 
In vLLM we should just assert the the returned logits have the expected vocab size.

Comment (tchedaTT) [reply vllm/v1/worker/tt_model_runner.py:1740]: +1, I think our contract should be that the logits are returned unpadded. If we need padding at the ttnn level, it should be removed in the generator before/after moving to torch.
